src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -127,17 +128,13 @@ private async Task ProcessAsyncActions(AsyncAction[][] actions)
                         case AsyncActionType.CreateTimer:
                             using (var cts = new CancellationTokenSource())
                             {
-                                DurableOrchestrationContext ctx = this.context as DurableOrchestrationContext;
-
                                 if (ctx != null)
                                 {
-                                    tasks.Add(ctx.OutOfProcCreateTimer(ctx, action.FireAt, cts.Token));
-                                }
-                                else
-                                {
-                                    tasks.Add(this.context.CreateTimer(action.FireAt, cts.Token));
+                                    ctx.ValidateOutOfProcTimer(action.FireAt);
```

Comment:
What about the timer does this validate? It would be good to add it as an inline comment :)

Reply:
I made the method name more clear, as opposed to a comment. I find comments get out of date, where method names are easier to refactor.